src/content/docs/references/docs/types.md

```diff
@@ -315,125 +315,184 @@ Distinct types is a kind of type alias which creates a new type that has the sam
+}
+```
 
 ### Generic types
+```c3
+import generic_list; // Contains the generic MyList
 
-    import generic_list; // Contains the generic MyList
+struct Foo { 
+    int x; 
+}
 
-    struct Foo { int x; }
+// ✅ def for each type used with a generic module.
+def IntMyList = MyList(<Foo>);
+MyListFoo working_example;
 
-    // Using def - usually recommended:
-    def IntMyList = MyList(<int>);
-    IntMyList abc;
+// ❌ An inline type definition will give an error.
+MyList<Foo> failing_example = MyList(<Foo>);
+```
+Read more about [generic types](/references/docs/generics).
 
-    // Inline type definition
-    MyList<Foo> bcd = MyList(<Foo>);
+## Enum
 
-Read more about generic types on [the page about generics](/references/docs/generics).
+Enum or enumerated types use the following syntax:
+```c3
+enum State : int 
+{
+    WAITING,
+    RUNNING,
+    TERMINATED
+}
 
-## Enum
+// Access enum values via:
+State current_state = State.WAITING;
+```
+The access requires referencing the `enum`'s name as `State.WAITING` because 
+an emum like `State` is a separate namespace by default, just like C++'s class `enum`.
 
-Enum (enumerated) types use the following syntax:
 
-    enum State : int 
-    {
-      PENDING,
-      RUNNING,
-      TERMINATED
-    }
+### Enum associated values
 
-Enum constants are namespaces by default, just like C++'s class enums. So accessing the enums above would for example use `State.PENDING` rather than `PENDING`.
+It is possible to associate each enum value one or more a static values.
+```c3
+enum State : int (String description) 
+{
+    WAITING = "waiting",
+    RUNNING = "running",
+    TERMINATED = "ended",
+}
 
-### Enum type inference
+fn void main() 
+{
+    State process = State.RUNNING;
+    io::printfn("%s", process.description);
+}
+```
+Multiple static values can be associated with an enum value, for example:
+```c3
+struct Position
+{
+    int x;
+    int y;
+}
 
-When an enum is used in where the type can be inferred, like in case-clauses or in variable assignment, it is allowed to drop the enum name:
+enum State : int (
+    String   desc,
+    bool     active,
+    Position pos,
+)
+{
+    WAITING    = { "waiting", false, { 1, 2} },
+    RUNNING    = { "running", true,  {12,22} },
+    TERMINATED = { "ended",   false, { 0, 0} },
+}
 
-    State foo = PENDING; // State.PENDING is inferred.
-    switch (foo)
+fn void main()
+{
+    State process = State.RUNNING;
+    if (process.active)
     {
-      case RUNNING: // State.RUNNING is inferred
-        ...
-      default:
-        ...
+        io::printfn("Process is: %s", process.desc);
+        io::printfn("Position x: %d", process.pos.x);
     }
+}
+```
 
-    fn void test(State s) { ... }
-
-    ...
-
-    test(RUNNING); // State.RUNNING is inferred
+### Enum type inference
 
+When an `enum` is used in where the type can be inferred, like switch case-clauses or in variable assignment, it is allowed to drop the enum name:
```

Comment:
"in where" -> "where"
"it is allowed to drop the enum name" -> "the enum name is not required"